database/mongodb/mongodb.go

```diff
@@ -80,7 +81,12 @@ func (m *Mongo) Open(dsn string) (database.Driver, error) {
 	migrationsCollection := unknown.Get("x-migrations-collection")
 	transactionMode, _ := strconv.ParseBool(unknown.Get("x-transaction-mode"))
 
-	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(dsn))
+	credential := options.Credential{
+		Username: uri.Username,
+		Password: uri.Password,
+	}
+	clientOpts := options.Client().ApplyURI("mongodb://" + strings.Join(uri.Hosts, ",")).SetAuth(credential)
```

Comment:
How's this approach better than the original?

Reply:
We don't have error "sasl conversation error: unable to authenticate using mechanism "SCRAM-SHA-1": (AuthenticationFailed) Authentication failed." on migration commands